LSPS5/README.md

```diff
@@ -0,0 +1,623 @@
+
+`webhook` is a required string, containing the URI of the
+webhook that the LSP can use to push a notification to the
+client.
+The `webhook` can be up to 1024 bytes in UTF-8 format, with
+the same rules as for `app_name`.
+
+The `webhook` MUST have one of the following protocols:
+
+* `https`
+
+The client:
+
+* MUST use a HTTPS-protocol URI to the notification delivery
+  service.
+* SHOULD encode the following data in any part of the URI
+  (hostname, path, or `GET` parameters):
+  * The LSP node ID.
+  * Some authorization to allow the given LSP to wake the
+    client, ideally a proof that the client has a channel
+    or other relationship with the LSP.
+  * Some way to identify the target device the client is
+    running on.
+  * This encoding MAY be done by registering all the information
+    with the notification delivery service and then getting a
+    token in return, and only putting the token in the URI.
+* MAY encode any additional information in the URI.
+
+> **Rationale** By placing most data in the URI, the LSP
+> interface is simplified and the client developer has the
+> flexibility to add any additional information they want
+> the LSP to simply pass through from the client instance
+> to the notification delivery service.
+
+If a webhook with the given `app_name` is already registered,
+then this call simply replaces the existing webhook and
+succeeds.
+Otherwise, the LSP MUST check if the number of registered
+webhooks is already at the limit it imposes.
+If it is still below the limit, the LSP inserts the new
+webhook under the given `app_name` and succeeds.
+(i.e. this is an "insert-or-replace" operation with a maximum
+number of inserted webhooks)
+
+The LSP:
+
+* MUST persist the given webhook and associate it with the client
+  node ID and given `app_name` before responding to this call.
+* MUST use **all** `webhook`s that were registered by successful
+  `lsps5.set_webhook` calls, ***except*** for the
+  `lsps5.webhook_registered` notification.
+* MUST remember all webhooks:
+  * If there are currently no channels with this client, for at
+    least 7 days.
+    * If a new channel with the client is opened, then as below.
+  * If there is at least one open channel with this client,
+    indefinitely remember them, until for at least 7 days after
+    the last channel with the client is closed.
+
+On success, the LSP also contacts the webhook and sends the
+`lsps5.webhook_registered` notification without parameters.
+Only the newly-registered webhook is contacted for this
+notification (unlike other events, which contact all
+registered webhooks).
+The LSP MAY send this notification before or after responding to
+the `lsps5.set_webhook` call.
+The LSP MUST send this notification to this webhook before sending
+any other notifications to this webhook.
+This `lsps5.webhook_registered` notification is sent, whether or
+not this is a new registration, even if the client re-registered
+the same `app_name` and `webhook`.
```

Comment:
Hmm okay.

Reply:
I want to go with returning a flag `no_change` in the result (`true` if the `app_name` and `webhook` exists and are the same as previous and the LSP DID NOT send `webhook_registered`, `false` otherwise and the LSP did/will send `webhook_registered`).  The reasoning is:

* The client really wants te `webhook_registered` as feedback that the path client->LSP->notification delivery service->client works.
* The LSP needs to inform the client if it is not using that path because it already has the same information and we do not want to waste bandwidth.

On the other hand ---- what if the client->LSP->notification delivery server->client path does NOT work? Note that the LSP gets no feedback on whether the notification delivery server actually is able to deliver the notification to the client --- the human user might have notifications for a client app disabled, and the information may not be sent back by the operating system of the device.  In that case, the client may want to ask the human user to re-enable notifications for the client app, then ask the LSP to try again.  Re-registering the same `app_name`+`webhook` was what I thought would implement this. Do we need this user story?

Reply:
Went with `no_change` in the result of `set_webhook` depending on whether the `app_name` and `webhook` are exactly the same as a previous call.

Reply:
closing.